ksql-engine/src/main/java/io/confluent/ksql/KsqlExecutionContext.java

```diff
@@ -88,17 +88,28 @@
    */
   PreparedStatement<?> prepare(ParsedStatement stmt);
 
+  /**
+   * Executes a statement using the engine's primary service context.
+   *
+   * @see #execute(ServiceContext, ConfiguredStatement)
+   */
+  ExecuteResult execute(ConfiguredStatement<?> statement);
+
   /**
    * Execute the supplied statement, updating the meta store and registering any query.
    *
    * <p>The statement must be executable. See {@link KsqlEngine#isExecutableStatement}.
    *
    * <p>If the statement contains a query, then it will be tracked, but not started.
    *
+   * <p>The statement is executed using the specific {@code ServiceContext}
+   *
+   * @param serviceContext The ServiceContext of the user executing the statement.
    * @param statement The SQL to execute.
    * @return The execution result.
    */
-  ExecuteResult execute(ConfiguredStatement<?> statement);
+  ExecuteResult execute(ServiceContext serviceContext, ConfiguredStatement<?> statement);
```

Comment:
I think one way of structuring this that would make the code easier to understand, and make it easier to guarantee that we're using the right context in the right places would be to pass a `Function<ServiceContext, KsqlExecutionContext> executionContextFactory` to all the API endpoints. That way we could drop this execute method, and there is no way for the callers to use the wrong context by accident.